src/worldstate/models/items/weapon.rs

```diff
@@ -90,18 +118,25 @@ pub struct Weapon {
 
     pub unique_name: String,
 
-    pub vaulted: bool,
+    /// This will be [Some], if [RangedWeapon::is_prime] is true
+    pub vaulted: Option<bool>,
```

Comment:
For ease of consuming would this be better or do we want to leave it as an Option? Just thinking because it's such a small amount of data anyway
```suggestion
    #[serde(default = "bool::default")]
    pub vaulted: bool,
```

`bool::default` is false but if we didnt want to take this approach and wanted to be more explicit we could also set the default to `default_bool::<false>` and use an extra util function like below:
```rust
pub const fn default_bool<const V: bool>() -> bool {
    V
}
```

Reply:
I don't really know.. I think it would be rather misleading if any state of "vaulted" would be on a non-prime weapon, whether true or false?

Reply:
That's valid, marking as resolved